pkg/cli/image/manifest/manifest.go

```diff
@@ -282,21 +304,30 @@ func FirstManifest(ctx context.Context, from imagereference.DockerImageReference
 	}
 
 	originalSrcDigest := srcDigest
-	srcManifests, srcManifest, srcDigest, err := ProcessManifestList(ctx, srcDigest, srcManifest, manifests, from, filterFn, false)
+	// Do not return a manifest list, filter here to return appropriate os/arch
+	if o.IsWildcardFilter() {
```

Comment:
I don't understand why this is here?  This seems unnecessary and is better handled outside this method.

Reply:
I removed the need for this error, removed this code